packages/ui/v2/core/layouts/SettingsLayout.tsx

```diff
@@ -89,18 +93,160 @@ const useTabs = () => {
+        </div>
+        {tabsWithPermissions.map((tab) => {
+          return tab.name !== "teams" ? (
+            <>
+              <div>
+                <div
+                  className="group flex h-9 w-64 flex-row items-center rounded-md px-3 py-[10px] text-sm font-medium leading-none text-gray-600 hover:bg-gray-100  group-hover:text-gray-700 [&[aria-current='page']]:bg-gray-200 [&[aria-current='page']]:text-gray-900"
+                  key={tab.name}>
+                  <tab.icon className="mr-[12px] h-[16px] w-[16px] self-start stroke-[2px] md:mt-0" />
+                  <p>{t(tab.name)}</p>
+                </div>
+              </div>
+              <div className="mt-2">
+                {tab.children?.map((tab) => (
+                  <VerticalTabItem
+                    key={tab.name}
+                    name={t(tab.name)}
+                    href={tab.href}
+                    textClassNames="px-3 text-gray-900 font-medium text-sm"
+                    disableChevron
+                  />
+                ))}
+              </div>
+            </>
+          ) : (
+            <>
+              <div>
+                <div
+                  className="group flex h-9 w-64 flex-row items-center rounded-md px-3 py-[10px] text-sm font-medium leading-none text-gray-600 hover:bg-gray-100  group-hover:text-gray-700 [&[aria-current='page']]:bg-gray-200 [&[aria-current='page']]:text-gray-900"
+                  key={tab.name}>
+                  <tab.icon className="mt-2 mr-[12px] h-[16px] w-[16px] self-start stroke-[2px] md:mt-0" />
+                  <p>{t(tab.name)}</p>
+                </div>
+                {teams &&
+                  teamMenuState &&
+                  teams.map((team, index: number) => (
+                    <Collapsible
+                      key={team.id}
+                      open={teamMenuState[index].teamMenuOpen}
+                      onOpenChange={() =>
+                        setTeamMenuState([
+                          ...teamMenuState,
+                          (teamMenuState[index] = {
+                            ...teamMenuState[index],
+                            teamMenuOpen: !teamMenuState[index].teamMenuOpen,
+                          }),
+                        ])
+                      }>
+                      <CollapsibleTrigger>
+                        <div
+                          className="flex h-9 w-64 flex-row items-center rounded-md px-3 py-[10px] text-sm font-medium leading-none hover:bg-gray-100  group-hover:text-gray-700 [&[aria-current='page']]:bg-gray-200 [&[aria-current='page']]:text-gray-900"
+                          onClick={() =>
+                            setTeamMenuState([
+                              ...teamMenuState,
+                              (teamMenuState[index] = {
+                                ...teamMenuState[index],
+                                teamMenuOpen: !teamMenuState[index].teamMenuOpen,
+                              }),
```

Comment:
Update the specific team menu based on the team's index in the initial team query